tests/federation_room_join_partial_state_test.go

```diff
@@ -182,6 +182,103 @@ func TestPartialStateJoin(t *testing.T) {
 			})
 		}
 	})
+
+	// test that a partial-state join can fall back to other homeservers when re-syncing
+	// partial state.
+	t.Run("PartialStateJoinSyncsUsingOtherHomeservers", func(t *testing.T) {
+		// set up 3 homeservers: hs1, hs2 and complement
+		deployment := Deploy(t, b.BlueprintFederationTwoLocalOneRemote)
+		defer deployment.Destroy(t)
+		alice := deployment.Client(t, "hs1", "@alice:hs1")
+		charlie := deployment.Client(t, "hs2", "@charlie:hs2")
+
+		// create a public room
+		roomID := alice.CreateRoom(t, map[string]interface{}{
+			"preset": "public_chat",
+		})
+
+		// create the complement homeserver
+		server := federation.NewServer(t, deployment,
+			federation.HandleKeyRequests(),
+			federation.HandlePartialStateMakeSendJoinRequests(),
+			federation.HandleEventRequests(),
+			federation.HandleTransactionRequests(
+				func(e *gomatrixserverlib.Event) {
+					t.Fatalf("Received unexpected PDU: %s", string(e.JSON()))
+				},
+				// hs1 may send us presence when alice syncs
+				func(e gomatrixserverlib.EDU) {},
```

Comment:
I think you can just write `nil` if there's nothing to do for EDUs.
```suggestion
				nil,
```